src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,105 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+/**
+ * Encapsulates a storage unit which handles loading Tasks from the
+ * file and saving Tasks in the file.
+ *
+ * @author Sean Chin Jun Kai
+ */
+public class Storage {
+    private static final String DIR_PATH = System.getProperty("user.home") + "/ip/data";
+    private String filePath;
+    public Storage(String path) {
+        this.filePath = path;
+    }
+
+    /**
+     * Loads all saved Tasks in txt file.
+     *
+     * @return List of saved Tasks.
+     * @throws DukeException if there is an error with reading or creating the txt file.
+     */
+    public TaskList load() throws DukeException {
+        TaskList tasks = new TaskList();
+        try {
+            File dir = new File(DIR_PATH);
+            dir.mkdirs();
+            File taskFile = new File(DIR_PATH + filePath);
+            if (!taskFile.exists()) {
+                taskFile.createNewFile();
+            }
+            Scanner fileScanner = new Scanner(taskFile);
+            while (fileScanner.hasNext()) {
+                String input = fileScanner.nextLine();
+                Task task = readTaskString(input);
+                tasks.addTask(task);
+            }
+        } catch (IOException | DukeException e) {
+            throw new DukeException(e.getMessage());
+        }
+        return tasks;
+    }
+
+    /**
+     * Saves all Tasks into txt file.
+     *
+     * @param tasks List of current Tasks.
+     * @throws DukeException if there is an error writing to the txt file.
+     */
+    public void saveToFile(TaskList tasks) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(DIR_PATH + filePath);
+            fw.write(tasks.saveString());
+            fw.close();
+        } catch (IOException e) {
+            throw new DukeException(e.getMessage());
+        }
+    }
+
+
+    /**
+     * Formats Tasks in txt file into corresponding Task objects.
+     *
+     * @param input Task in txt file.
+     * @return Task object.
+     * @throws DukeException if there is an error reading the file.
+     */
+    public static Task readTaskString(String input) throws DukeException {
```

Comment:
This method could be under Parser, I think even though it is under the context of the text file it is still a parser method.